### PR TITLE
switch messaging service type db enum to use lowercase values

### LINKED ISSUE #2680: consider switching to use lower-cased values for MessagingServiceType enum in db
### Is your feature request related to a specific problem?

Because the `MessagingServiceType` enum has all upper-cased values (both in our pydantic model and as defined in the db), but we also generally try to support API users inputting lower-cased values (whether it's in the case of a URL path or as fields in a request body), we have to keep some workarounds in place in our codebase.

### Describe the solution you'd like

It seems preferable that we'd just switch the enum (both in the pydantic model and in the db) to just have lowercased values, so we could remove a lot of the code that does the conversions.

This would require some relatively minor code updates and a db migration that may be a bit more complex. We'd remove some tech debt though and make things simpler for this enum moving forward.

### Describe alternatives you've considered, if any

Keep things the way they are

### Additional context

This came up when discussing the enum changes ultimately made in https://github.com/ethyca/fides/pull/2676- we decided not to take this on and do the simpler change there to support lower-cased values in URLs

### PR DESCRIPTION
**Pull request by adamsachs**
Closes #2680 

### Code Changes

* update `MessagingServiceType` Enum in python codebase to have lowercased values, update any references in code and in tests
    * switch any `.upper()` calls that coerced user-input values to instead be `.lower()`, to maintain backward compatibility and continue to support uppercased user input
* provide a DB migration to change the enum type in the DB schema, as well as a data migration to change any existing enum values in the DB to lowercased
* update FE code references to user lowercased enum values rather than uppercased

### Steps to Confirm

* ensure new system config UI still works as expected
* ensure migration works as expected:
    * run server on `main` (e.g. `nox -s dev -- ui`) and create a messaging config (e.g. using the system config UI)
    * teardown but do not delete volumes, i.e. `nox -s teardown`
    * checkout this branch, run server (e.g. `nox -s dev -- ui`) and ensure your messaging config is still present and functional (can check using the API - `GET /api/v1/messaging/default/{service_type}`, or checking the `messagingconfig` db table directly)

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [x] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### Description Of Changes

- Caveat: there's a very narrow edge case where if someone in the past few days has used the UI on `main` to configure a messaging config, the UI will have set the `notification.notification_service_type` property to an uppercased value, e.g. `MAILGUN`. 
    - you may see some`500`s on calls to `GET /api/v1/messaging/default/active` and that's what is used by the messaging config UI to load the "current" messaging provider. so the messaging config UI will show up as if you have no messaging provider configured
    - if this happens, then just click on the messaging provider you had configured in the messaging config UI and its details will be loaded. you should be good to go now.

### CODE REVIEW

**Review by eastandwestwind — APPROVED** [review]

File: src/fides/api/ops/models/messaging.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def get_by_type(
         """
         Retrieve the messaging config of the given type
         """
-        return db.query(cls).filter_by(service_type=service_type).first()
+        return db.query(cls).filter_by(service_type=service_type.value).first()
```
Comment (seanpreston) [inline src/fides/api/ops/models/messaging.py:130]: `sqlalchemy` must be auto-translating enums in the background for us for this to have worked originally?

File: src/fides/api/ops/service/messaging/messaging_crud_service.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ def create_or_update_messaging_config(
     data = {
         "key": config.key,
         "name": config.name,
-        "service_type": config.service_type,
+        "service_type": config.service_type.value,
```
[the remark this answers is not in the document]
Comment (seanpreston) [reply src/fides/api/ops/service/messaging/messaging_crud_service.py:32]: `sqlalchemy` does some [auto-translation of enums in the background](https://docs.sqlalchemy.org/en/14/core/type_basics.html#sqlalchemy.types.Enum), so either should work just fine. It's just that if `.value` is passed in it will validate it as a string (because that's what `.value` is).

**Review by seanpreston — APPROVED** [review]

File: src/fides/api/ops/service/messaging/messaging_crud_service.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (adamsachs) [reply src/fides/api/ops/service/messaging/messaging_crud_service.py:32]: > hm, was this broken before?

a good callout. i don't think it was.

my best guess is that i'd made this change _before_ i switched the variable names in the enum to lowercased (but after updating the enum values in the DB), and i was trying to make things work in that state. if that were the case, the DB wouldn't accept the uppercased variable names after i'd switched the db enum to lowercased - so we'd _have_ to use the `.value`.

but then i think there were other problems when trying to keep the variable names uppercased that motivated me to just switch them to lower case. and i kept these changes because they'd been in place and they seemed to make sense anyways (i.e., didn't break things or make the code less readable).

that being said, this doesn't really align with the functionality @seanpreston describes with `sqlalchemy` taking the `.value`. so the explanation may be wrong.

in summary, i'm not sure what motivated the change. but at this point, since we agree it doesn't break anything and isn't any less readable, we'll just keep it. sorry for the distraction on this one!
